aspnet/tutorials/mvc-with-entity-framework.rst

```diff
@@ -0,0 +1,394 @@
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Models/Author.cs
+    :language: c#
+
+Repeat these steps to add another class named ``Book`` with the following code:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Models/Book.cs
+    :language: c#
+
+To keep the app simple, each book has a single author. The ``Author`` property provides a way to navigate the relationship from a book to an author. In EF, this type of property is called a *navigation property*. When EF creates the database schema, EF automatically infers that ``AuthorID`` should be a foreign key to the Authors table. 
+
+Add Scaffolding
+^^^^^^^^^^^^^^^
+Scaffolding saves you time and coding effort by automatically generating the starting point for your application's CRUD (Create, Read, Update and Delete) operations. Starting from a simple model class, and, without writing a single line of code, you will create two controllers that will contain the CRUD operations related to books and authors, as well as the all the necessary views. 
+
+To add a scaffolding, right-click the **Controllers** folder in **Solution Explorer**. Select **Add** –> **New Scaffolded Item**.
+
+.. image:: mvc-with-entity-framework/_static/15a-scaffold-menu.png
+
+From the **Add Scaffold** dialog box, select **MVC 6 Controller with views, using Entity Framework**, then click the **Add** button.
+
+.. image:: mvc-with-entity-framework/_static/15b-add-scaffold-db.png
+
+Next, in the **Add Controller** dialog box, set the model class dropdown to **Book (ContosoBooks.Models)**. Also, set the data context class to **ApplicationDbContext (ContosoBooks.Models)**. Make sure the **Generate views** checkbox is checked. Then click the **Add** button.
+
+.. image:: mvc-with-entity-framework/_static/15c-book-controller.png
+
+As you can see in the above image, the **Add Controller** dialog box gives you the opportunity to select options for generating the controller and views. 
+
+This scaffold creates the code that provides a controller and a set of views. The views provide the UI and code to create, read, update, delete, and list data from the database.
+
+Repeat the above scaffolding steps to create an **Author** controller and related views. Use the **Author (ContosoBooks.Models)** model class and the **ApplicationDbContext (ContosoBooks.Models)** data context class as shown in the following image.
+
+.. image:: mvc-with-entity-framework/_static/15d-author-controller.png
+
+In the **Solution Explorer** you'll see that the new controllers were added within the **Controller** folder and new views were created within the **Views** folder.
+
+.. image:: mvc-with-entity-framework/_static/15e-updated-solution-explorer.png
+
+Add sample data
+^^^^^^^^^^^^^^^
+
+Rather than entering several sample records by hand, you will add code that will by used to populate your database. Add a class named ``SampleData`` in the *Models* folder with the following code:
```

Comment:
This is the only crucial bug that needs to be fixed. Code doesn't work without DI added first.